src/globals.rs

```diff
@@ -23,6 +23,7 @@ lazy_static! {
             // still be live on other threads on program exit
             ::hdf5_sys::h5::H5dont_atexit();
             ::hdf5_sys::h5::H5open();
+            ::hdf5_sys::h5e::H5Eset_auto2(::hdf5_sys::h5e::H5E_DEFAULT, None, core::ptr::null_mut());
```

Comment:
The question is now though, do we want to provide a way to insolence errors? Perhaps a fancy mutex counter thing is not needed anymore, more like a simple global 'silence_errors(true/false)`?